self_hosted/tokenizer.jou

```diff
@@ -209,6 +209,56 @@ struct Tokenizer:
                 self.unread_byte(c)
                 return level
 
+    # Assumes the initial ' has been read.
+    def read_byte_literal(self: Tokenizer*) -> byte:
+        c = self.read_byte()
+        if c == '\'':
+            fail(self->location, "a byte literal cannot be empty, maybe use double quotes to instead make a string?")
+        if c == '\0' or c == '\n':
+            if c == '\n':
+                self->location.lineno--
+            fail(self->location, "missing ' to end the byte literal")
+
+        if c == '\\':
+            after_backslash = self.read_byte()
+            if after_backslash == '\0':
+                fail(self->location, "missing ' to end the byte literal")
+            elif after_backslash == 'n':
+                c = '\n'
+            elif after_backslash == 'r':
+                c = '\r'
+            elif after_backslash == 't':
+                c = '\t'
+            elif after_backslash == '\\':
+                c = '\\'
+            elif after_backslash == '\'':
+                c = '\''
+            elif after_backslash == '"':
+                fail(self->location, "double quotes shouldn't be escaped in byte literals")
+            elif '0' <= after_backslash and after_backslash <= '9':
+                c = after_backslash - '0'
+            elif after_backslash < 0x80 and isprint(after_backslash) != 0:
+                message: byte* = malloc(100)
+                sprintf(message, "unknown escape: '\\%c'", after_backslash)
```

Comment:
I wonder if this runs when there is a backslash at the end of a line.

Reply:
I fixed it, but it's difficult to test automatically: If I add a `# Error` comment at the end of the line, then the backslash is no longer at the end of a line.